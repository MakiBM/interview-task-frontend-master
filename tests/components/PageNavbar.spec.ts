import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import Component from '../../src/components/PageNavbar.vue'
import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory(),
  routes: [
    { path: '/', component: { template: '<div>Home</div>' } },
    { path: '/stops', component: { template: '<div>Stops</div>' } },
  ],
})

describe('PageNavbar Snapshot Tests', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(Component, {
      global: {
        plugins: [router],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
