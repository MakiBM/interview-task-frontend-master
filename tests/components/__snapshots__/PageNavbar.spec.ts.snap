// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNavbar Snapshot Tests > matches the snapshot 1`] = `
"<div class="card border-0 mb-3 px-2 font-inter" data-v-e4f7cc59="">
  <nav class="card-body navbar navbar-expand navbar-light" data-v-e4f7cc59="">
    <div class="navbar-nav" data-v-e4f7cc59=""><a href="/" class="nav-link position-relative" data-v-e4f7cc59="">Bus Lines</a><a href="/stops" class="nav-link position-relative" data-v-e4f7cc59="">Stops</a></div>
  </nav>
</div>"
`;
